"Why RSE" blogpost based on NRSE conference discussion (#457)
nordic-rse/nordic-rse.github.io: content/blog/2024-06-20-why-rse.md

+++
title = "Small steps towards sustainable and open research"

[extra]

> author = "Aurélie Vancraeyenest, Samantha Wittke"

Author: ```suggestion
author = "Aurélie Vancraeyenest, Samantha Wittke, Konstantin Stadler"
```
@konstantinstadler please confirm if you want to be listed here as an author :)

Reviewer: Thanks for that! Very much like that and fully support the points of the blog post

**The short answer** 
People striving for "Better software, better research" 

**The long answer**
*Definition of Research Software ? (from Martin's slides)*
Research software is software that one use to generate, process or analyse results that is intended to appear in a publication. It can be anything from a few lines of code, to a professionally developed software package.

A Research Software Engineer is a person who combines an intricate understanding of research with professional software engineering expertise. RSE often have background in academia and have specialized in computing related areas, be it either software development, data analysis, (high-performance) computing, etc.

Independently of their area of expertise, their common goal is better maintainability, sustainability of research software and better reproducibility of research outputs.

**What an RSE is not**

Even though they are in close collaboration with IT services, RSEs are not IT support people, as in IT manager or IT system specialists. They have a dual background (research/coding), they take part in research projects and speak the "same language" as researchers. 

> They also are not enforcers, they don't enforce technologies or methods but build on the tools already in use in research groups.  

Reviewer: Disagree a bit on this. We (talking of my group) are actually "enforcers". Through collaboration with us we actually enforce use of git, some code style, etc. So while just question style advice is of course free from that, as soon as we collaborate with anyone we setup a github repo, change to some workable project setup etc. 

There might be a difference between RSE groups for larger institutions and small RSE groups like ours here, where we can serve at a more detailed level.

Reviewer 2: Yes, we can soften. I assume what attendees meant here is that RSEs very much build on tools/tech researchers already know and use, but of course it shouldn't mean that RSEs don't also introduce best practice tools. I think Samantha's suggestion goes along the right direction.

Author: ```suggestion
They may introduce best practice tools and techniques to a research group by showing the benefits and teaching them how to use them independently. But they can also estimate if a tool that the research group is used to is good enough for the job at hand.
```

Let's iteratively improve this sentence(s) :)

Reviewer 2: After further thinking on this and re-reading through the workshop notes, I am wondering if it's not best to simply remove this sentence completely. It's not clear from the notes what the attendees meant here by "RSE are not enforcers" and by guessing or interpreting, we might end-up distorting what the initial message was. So I would suggest to just remove this sentence.

Author: True. Should we still keep something about "finding right tool for the job" ?

Reviewer 2: Sure, let's then move this part a bit above. I'll add a suggestion based on yours.

Reviewer: Good for me

#### Why would an organization need RSEs? - Current state of Research software 

Does any of following scenario sound familiar? 

* A PhD/post doc has developed some really nice piece of code, but they left and no one knows anymore how to run it or develop it further. 
* Someone wrote some code long time ago that is used daily by a research group, but no one knows the technology in the research group anymore and it cannot be modified, updated or compiled anymore. 
* When a newcomer joins a team and starts familiarizing with analysis/modelling/etc tools, the only way is to "ask others who use it". No document or tutorials are available. 
* Some essential piece of software runs on a very old machine or on very old version of language and it's very difficult to replace or update, if even possible. This can even lead as far as stopping research activities in the team for some time 
* After an article is submitted, one reviewer ask for additional analysis, or slightly different plots, but upon running the code again, one cannot reproduce the plots/results anymore. 

#### How RSE can help ? 

**Short answer**

> Improve FAIRness of research software and by that save time and resources. 

Reviewer: not quite sure if this is the angle we should take. Having all on board on "FAIR" is wishful thinking and I think RSE is actually broader than FAIR (talking basically of Open Science here). It is perfectly feasible for RSE to help on internal or closed code which will never be published/distributed. 

In my understanding this should become a piece which you can hand to any Prof/Leader, so (s)he should also be on board if they are not sigining up on the FAIR thing.

So perhaps: Improve the (re)usability of research software - including improving performance, reduce time and resource usage and ease on-boarding of new users.

Author: ```suggestion
Improve the (re)usability of research software - including improving performance, reduce time and resource usage and ease on-boarding of new users.
```
Agree with this suggestion

**Long answer**

RSE can leverage all the above issues and more by accompanying researchers in their software development journey. RSEs' technical knowledge and creativity on solving problems allow seeing where researchers would benefit the most. Here are a few examples/use cases: 

* Researchers have great ideas, but might not know how, or have time to develop the corresponding tools 

> * Help to get rid of repetitive or time consuming steps (that users don't want to do themselves) by introducing automatization 

Author: ```suggestion
* Help to get rid of repetitive or time consuming steps (that users don't want to do themselves) by introducing automatization 
* RSEs know about the FAIR principles and can apply them to research software
```

* Write documentation so the information doesn't get lost when people leave
* Make the regular technology updates to prevent that it become obsolete and break
* Supporting when switching from laptop to compute clusters 
* Many groups need the same thing, RSEs know how to create flexible templates to simplify e.g. spinning up a website to present research outputs

> * … what other good examples? 

Reviewer: I would rather bring FAIR software in here (+ publication of software) as one point here

Author: Suggestion added above

They can also act as intermediate in technical discussions about computing needs, as they know the needs of researchers since they are involved in research in a daily basis, and are also technology-aware people. 

As a summary, researchers with very deep expertise and RSEs with shallower but wider range of interests and skills complement each other very efficiently. 

#### How organisations can help?

Many organisations already have people doin RSE work, but the people and the organisation might not be aware of the term RSE. To give those people a sense of belonging and knowing that there is others like them, organisations could:

> 1) Support community involvement and self development
>     - Give people room to be active in RSE communities, attend conferences and trainings beyond academic work.
> 
> 2) Spread the word - be an RSE ambassador
>     - Recognize and highlight RSE (often behind the scenes and non-academic) work
>     - Invite RSEs to talk about their work
>     - Support RSEs building their own network within the organisation (connect them to administration, other RSEs, researchers, etc)
> 
> 3) Make it a THING (administrative existance)
>     - Hire RSEs
>     - If you have them already, gather them ("virtual team" or central RSE group)
>     - Offer support to spread good enough software practices in research groups
>     - Train the trainer events are important and needed 
>     - Support formation of RSE groups by providing guidance about practices for forming a new entity in the organisation (group, service, etc) and finding the right integration points
>     - Share organizational knowledge on how get internal funding for RSE-type work
>     
> 4) Funding
>     - Inlcude RSEs at the stage of grant proposal writing
>     - Provide central RSE funding to support smaller projects and internal tool developments 
>     - RSE benefits from open ended, non temporary funding - like other techical supports
>     - training and collaboration funding (e.g. EuroCC, EOSC)

Author: ```suggestion
1) Support community involvement and self development
    - Give people room to be active in RSE communities, attend conferences and trainings beyond academic work.

2) Acknowlege RSE work
    - Do not solely focus on authorship when evaluating the scientific output
    - While the contribution of the RSE team/engineer might not be enough for a co-authorship, an acknowledgement in the paper can provide an adequate way to honor the work provided by RSEs
    - Introduce this practice in your organizational guidelines and courses
    - Provide example statements to acknowledge the work behind the scenes (e.g. "Code development and data management were supported by [names of your friendly RSEs or the RSE group].")
 
3) Spread the word - be an RSE ambassador
    - Recognize and highlight RSE (often behind the scenes and non-academic) work
    - Invite RSEs to talk about their work
    - Support RSEs building their own network within the organisation (connect them to administration, other RSEs, researchers, etc)

4) Make it a THING (administrative existance)
    - Hire RSEs
    - If you have them already, gather them ("virtual team" or central RSE group)
    - Offer support to spread good enough software practices in research groups
    - Train the trainer events are important and needed 
    - Support formation of RSE groups by providing guidance about practices for forming a new entity in the organisation (group, service, etc) and finding the right integration points
    - Share organizational knowledge on how get internal funding for RSE-type work
    
5) Funding
    - Inlcude RSEs at the stage of grant proposal writing
    - Provide central RSE funding to support smaller projects and internal tool developments 
    - RSE benefits from open ended, non temporary funding - like other techical supports
    - training and collaboration funding (e.g. EuroCC, EOSC)
```

Please edit as you see fit :)

Reviewer 2: This is a really nice addition, I like it as it is

> 

Reviewer: 5) acknowledge them.
While the contribution of the RSE team/engineer might not be enough for a co-author ship, acknowledge the role for RSE in your analysis in the acknowledgment of the article. At the digital lab we actually provide two ready statements for this:

For coding/data help:
"Code development and data management [DELETE AS NECESSARY] were supported by the Industrial Ecology Digital Laboratory."

For access to the servers:
"The computations/simulations/[SIMILAR] were performed on resources provided by the Industrial Ecology Digital Laboratory."

Author: Oh yes! This is indeed missing, and a step that does not "cost" organizations anything. 
Should this actually be number 1 or 2 rather than 5?

Reviewer 2: Excellent suggestion! I would put it at the same level as 2)

Author: added a suggestion to the text, pleas edit as you see fit

#### Get deeper

> 

Reviewer: article on the setup of a smaller RSE group:
https://zenodo.org/records/997904

Author: Added as a resource. 
Do @bast , @rkdarst have any reading materials/links to add here about your experiences?

Reviewer 2: Are we expecting more resources to be linked or were they all added already?

Author: Added Aalto RSE history.

Does RSE sound like something you would need in your organisation, or want to hear more about what RSEs in the Nordics are about, please reach out to us at [Nordic-RSE](https://nordic-rse.org/).

Other useful resources: 

> - Martins slides https://zenodo.org/doi/10.5281/zenodo.11399991

Author: ```suggestion
- [Slides of the Nordic-RSE keynote by Martin O'Reilly on "The growth of Research Software Engineering and its vital role in research"](https://zenodo.org/doi/10.5281/zenodo.11399991)
```
